Roll iotam and iota deployments if config has changed

modularTaco opened the pull request:
- Added a configmap hash to IOTA and IOTAM deployments so the deployments get rolled if the config changes
- Bumped IOTAM version in Readme and values.yaml to the latest stable one (works fine)
- Replaced the mongo host in the values of the IOTAM chart with the service name, as since a few versions the IOTAM can connect to a mongo replicaset without specifying each member seperately
- replaced deployment apiVersion `apps/v1beta1` with `apps/v1` in autoscaler

chicco785 commented:
#97 will fix the build mechanism, once merged, you should rebase if everything works fine, we will then merge ;)

c0c0n3 commented:
@modularTaco #97 just got merged, if you could please rebase on master, thanks alot!!

c0c0n3 commented:
@modularTaco thanks for your pr, that's great. the helm linter is moaning about some missing newlines at the end of some files---see linter errors. if you could please fix that, thanks soooo much!

modularTaco commented on charts/iot-agent-manager/values.yaml:
```diff
@@ -37,7 +37,7 @@ mongo:
   rs: rs
   # The hostname of the mongodb service
-  dbhost: mongo-rs-mongodb-replicaset-0.mongo-rs-mongodb-replicaset.prod,mongo-rs-mongodb-replicaset-1.mongo-rs-mongodb-replicaset.prod,mongo-rs-mongodb-replicaset-2.mongo-rs-mongodb-replicaset.prod
+  dbhost: mongo-rs-mongodb-replicaset.prod
```
@c0c0n3 Unfortunately i'm not using IOTA and IOTAM anymore and this PR is quite old, so i do not remember completely. But there is something in the back of my mind that this change was not as flawless as it should be and we ran into some problems with it. Could you please doublecheck that using the servicename instead of the members seperately is supported and works fine with mongo pod restarts?

modularTaco commented:
Hi @c0c0n3 , i've fixed the file endings so the linter shouldn't complain anymore. But while doing so i read through the mongo uri change and remembered some problems with it. I've added a [comment](https://github.com/orchestracities/charts/pull/93/files#r925684182), could you please check prior merging?

c0c0n3 commented:
hi @modularTaco 

> fixed the file endings so the linter shouldn't complain anymore. 

yep, it looks like everything is hunky-dory now, fan-tas-tic!!

> read through the mongo uri change and remembered some problems with it. ... could you please check prior merging?

no prob, I've seen your comment. we'll review and retest to see if there are any issues.

thanks alot!
